interpreter/dotnet/instance.go

```diff
@@ -167,23 +158,24 @@ type dotnetInstance struct {
 	moduleToPEInfo map[libpf.Address]*peInfo
 
 	addrToMethod *freelru.LRU[libpf.Address, *dotnetMethod]
-
-	symbolizedLRU *freelru.LRU[symbolizedKey, libpf.Void]
 }
 
 // calculateAndSymbolizeStubID calculates a stub LineID, and symbolizes it if needed
-func (i *dotnetInstance) calculateAndSymbolizeStubID(symbolReporter reporter.SymbolReporter,
-	codeType uint) {
-	if i.codeTypeMethodIDs[codeType] != 0 {
-		return
-	}
+func (i *dotnetInstance) insertAndSymbolizeStubFrame(symbolReporter reporter.SymbolReporter,
+	trace *libpf.Trace, codeType uint) {
 	name := "[stub: " + codeName[codeType] + "]"
-	h := fnv.New128a()
-	_, _ = h.Write([]byte(name))
-	nameHash := h.Sum(nil)
-	lineID := libpf.AddressOrLineno(npsr.Uint64(nameHash, 0))
-	i.codeTypeMethodIDs[codeType] = lineID
-	symbolReporter.FrameMetadata(stubsFileID, lineID, 0, 0, name, "")
+	lineID := i.codeTypeMethodIDs[codeType]
+	if lineID == 0 {
+		h := fnv.New128a()
+		_, _ = h.Write([]byte(name))
+		nameHash := h.Sum(nil)
+		lineID = libpf.AddressOrLineno(npsr.Uint64(nameHash, 0))
```

Comment:
Does it make sense to have an extra function for this?

Reply:
The same pattern is written out in several interpreter modules. It might make sense to make a helper for this in the `interpreter` or `libpf` module. But I'd make it a separate / follow up PR instead since its logically a separate thing.